rs/nns/sns-wasm/canister/canister.rs

```diff
@@ -64,23 +64,24 @@ impl CanisterApi for CanisterApiImpl {
     }
 
     /// See CanisterApi::create_canister
-    async fn create_canister(
```

Comment:
So putting the logic at this level means the logic is now outside of unit tests that mock the CanisterAPI.

The logic for deciding the memory limits and everything else should be able to be under unit test coverage.  This method should be as dumb as possible.

Reply:
I would maybe make this method take the CanisterSettingsArgs or just the wasm_memory_limit (which could be optional with a default)

Reply:
Done!